endpoints/citrus-http/src/main/java/org/citrusframework/http/controller/HttpMessageController.java

```diff
@@ -130,6 +134,7 @@ private ResponseEntity<?> handleRequestInternal(HttpMethod method, HttpEntity<?>
 
         HttpServletRequest servletRequest = ((ServletRequestAttributes) attributes).getRequest();
         UrlPathHelper pathHelper = new UrlPathHelper();
+        pathHelper.setRemoveSemicolonContent(!endpointConfiguration.isHandleSemicolonPathContent());
```

Comment:
Instead of negating the setting maybe we should also use `isRemoveSemicolonPathContent()` on the endpoint and set this to `true` by default.